src/sub_system.mli

```diff
@@ -0,0 +1,11 @@
+open! Import
+
+include module type of struct include Sub_system_intf end
+
+module Register_backend(M : Backend) : Registered_backend with type t = M.t
+
+module Register_with_backend(M : With_backend) : sig end
```

Comment:
Could you leave a comment for each of these functors?

Reply:
I added some doc. I also renamed `With_backend` as it wasn't a really good name